### PR TITLE
Publicize: show options only to roles that can change their settings fixes #3126 fixes #3132

### LINKED ISSUE #3126: "Publicize: Not Connected" should not be displayed to users who can't change this setting
The "Publicize: Not Connected" text in the "Publish" meta box whilst editing a post should not be displayed if the user is unable to edit the settings to correct it.

Here is a screen grab as a Contributor:
image

### PR DESCRIPTION
**Pull request by eliorivero**
Returns early if current user doesn't have the `publish_posts` capability, available by default only to authors, editors and administrators.

### CODE REVIEW

File: modules/publicize/ui.php [hunk 1]
```diff
@@ -14,6 +14,11 @@ class Publicize_UI {
 	* Hooks into WordPress to display the various pieces of UI and load our assets
 	*/
 	function __construct() {
+		// Show only to users that can manage their settings.
+		if ( ! current_user_can( 'publish_posts' ) ) {
```
Comment (zinigor) [inline modules/publicize/ui.php]: Let's check for `$this->publicize->GLOBAL_CAP` instead after the property becomes available?

Comment (eliorivero) [inline modules/publicize/ui.php]: Added in dd88881

**Comment by chaselivingston** [comment]
@eliorivero It looks like this may have also removed the ability for some users to edit custom Publicize messages for posts, was this intentional?

**Comment by eliorivero** [comment]
@chaselivingston yes, only users with `Publicize->publicize->GLOBAL_CAP` capability will be able to manage Publicize.
Worth noting that Publicize in WP Admin will mostly disappear in 4.0 when users will be pointed to Calypso to manage everything as pointed in #3258

**Pull request merged**